languages/csharp/exercises/concept/nullability/.docs/introduction.md

```diff
@@ -0,0 +1,102 @@
+Sometimes, we need to indicate that a variable has no value.  In C#,
+we use the `null` literal to denote the absence of a value.  A
+*nullable* type is a type that allows `null` values.  The operator `?`
+can be added as a suffix to a type to make it nullable.
+
+In the following example, we declare a regular, non-nullable variable
+`a` of type `int` and a nullable variable `b` of type `int?` whose
+value is `5`:
+
+```csharp
+int a = 4;
+int? b = 5;
+```
+
+`a` must contain an integer value, while `b` can also be set to
+`null`:
+
+```csharp
+a = null; // Compile error as a is not nullable
+a = 5;    // Valid
+
+b = null; // Valid
+b = 6;    // Valid
+```
+
+A common mistake is trying to call a method or set/get a property on a
+variable whose value is `null`.  This will cause the C# runtime
+to raise an exception called `NullReferenceException`.
+
+The C# compiler can help us avoiding such errors. If we try to
```

Comment:
Having read this document again, I think it would be good to at this point introduce the difference between value and reference types: their nullability defaults. We can mention that in C# 8, the default can be changed (opt-in). Could you rewrite the following section a bit? I could also do that if you'd prefer that.

Reply:
I am not going to introduce the difference between reference and value types here right? 
Just the nullability defaults.

Reply:
Just pushed something

Reply:
> I am not going to introduce the difference between reference and value types here right?

Correct! This exercise can take a dependency on the `memory-allocation` and thus assume that the student knows about value and reference types.

Reply:
Ok, I will add the dependency